src/mpid/ch4/shm/posix/posix_send.h

```diff
@@ -16,12 +16,26 @@
 
 #include "posix_impl.h"
 
+/* Common macro used by all MPIDI_POSIX_mpi_send routines to compute vnis */
+#define MPIDI_POSIX_SEND_VNIS(vni_src_, vni_dst_) \
+    do { \
+        if (*request != NULL) { \
+            /* workq path */ \
+            vni_src_ = 0; \
+            vni_dst_ = 0; \
+        } else { \
+            vni_src_ = MPIDI_OFI_get_vni(SRC_VCI_FROM_SENDER, comm, comm->rank, rank, tag); \
+            vni_dst_ = MPIDI_OFI_get_vni(DST_VCI_FROM_SENDER, comm, comm->rank, rank, tag); \
+        } \
+    } while (0)
+
```

Comment:
I have doubt here that whether it is allowed or not to access `ofi` layer functions from inside the `posix` layer, although, it runs without error.

Reply:
The `MPIDI_OFI_get_vni()` function has been duplicated as `MPIDI_POSIX_get_vni()` inside the `posix`.

Reply:
> 
> 
> I have doubt here that whether it is allowed or not to access `ofi` layer functions from inside the `posix` layer, although, it runs without error.

No, it is not allowed (by policy). Just create/duplicate a `MPIDI_POSIX_get_vni` function.